libs/shared/modules/src/application/application.service.ts

```diff
@@ -120,11 +124,35 @@ export class ApplicationService implements IApplicationService {
   async getPrice(
     applicationId: string,
   ): Promise<ResultWrapper<CasePriceResponse>> {
-    ResultWrapper.unwrap(await this.getApplication(applicationId))
+    /**
+     * First we check if there is an existing case for the application.
+     * If so then we return the price of the case.
+     * If not then we calculate the price of the application.
+     */
+    try {
+      const caseLookup = (
+        await this.utilityService.caseLookupByApplicationId(applicationId)
+      ).unwrap()
 
-    return ResultWrapper.ok({
-      price: calculatePriceForApplication(),
-    })
+      /**
+       * If price has not been set we return default price
+       */
+
+      const price = caseLookup.price ? caseLookup.price : DEFAULT_PRICE
+
+      return ResultWrapper.ok({
+        price: price,
+      })
+    } catch (error) {
```

Comment:
Er `catch` aðallega fyrir lookup? Skilum við þá alltaf verði þó að engin application fannst?

Reply:
Já núna eru lookup föllinn stillt þannig að þau throwa not found ef lookupið fannst ekki. En alltaf hægt að breyta því ef við viljum. Þess vegna wrappa ég þetta fall í try catch.